Sources/Actions/BackwardCompatibility.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Simple Machines Forum (SMF)
+ *
+ * @package SMF
+ * @author Simple Machines https://www.simplemachines.org
+ * @copyright 2023 Simple Machines and individual contributors
+ * @license https://www.simplemachines.org/about/smf/license.php BSD
+ *
+ * @version 3.0 Alpha 1
+ */
+
+namespace SMF\Actions;
+
+use SMF\BackwardCompatibility as BackCompat;
+
+trait BackwardCompatibility
+{
+	use BackCompat;
+
+	/**
+	 * Called by Subs-Compat.php BackwardCompatibility wrapper functions to provide subaction
+	 * execution for existing mods
+	 *
+	 * @param null|string $sa
+	 * @param bool $return_config
+	 * @return null|array
+	 */
+	public static function subActionProvider(?string $sa = null, bool $return_config = false, ?string $activity = null): ?array
+	{
+		if ($return_config) {
+			return self::getConfigVars();
+		}
+
+		self::load();
+
+		if (is_string($sa)) {
+			// make sure its lowecase
+			if (array_key_exists($sa, self::$subactions)) {
```

Comment:
Based on the comment, I assume you meant to do this:

```suggestion
			// Make sure it's lowercase.
			$sa = strtolower($sa);
			
			if (array_key_exists($sa, self::$subactions)) {
```

Reply:
Actually that should have been removed because we can't normalize it since there is like 3-4 subactions that are not all lowercase. Seems like every time I try to normalize based of something always being done a certain way... wrong answer cause there is always just a handful of cases that violates it.

Reply:
updated the comment to reflect actual usage which is to verify its a supported subaction.